devices/DS18B20TemperatureSensor/main.c

```diff
@@ -0,0 +1,227 @@
+/***** XBEE APPLICATION PROJECT *****
+ * 
+ * Auto-generated header with information about the 
+ * relation between the XBee module pins and the 
+ * project components.
+ * 
+ ************ XBEE LAYOUT ***********
+ * 
+ * This layout represents the XBee ZB (S2CTH) module 
+ * selected for the project with its pin distribution:
+ *               _________________
+ *              /     ________    \ 
+ *             /     |   __   |    \ 
+ *            /      | //  \\ |     \ 
+ *   XPIN1  -|       | \\__// |      |- XPIN20
+ *   XPIN2  -|       |________|      |- XPIN19
+ *   XPIN3  -|                       |- XPIN18
+ *   XPIN4  -| ===================== |- XPIN17
+ *   XPIN5  -| #   # ####  #### #### |- XPIN16
+ *   XPIN6  -|  # #  #   # #    #    |- XPIN15
+ *   XPIN7  -|   #   ####  ###  ###  |- XPIN14
+ *   XPIN8  -|  # #  #   # #    #    |- XPIN13
+ *   XPIN9  -| #   # ####  #### #### |- XPIN12
+ *   XPIN10 -| ===================== |- XPIN11
+ *           |_______________________|
+ * 
+ ************ PINS LEGEND ***********
+ * 
+ * The following list displays all the XBee Module pins 
+ * with the project component which is using each one:
+ * 
+ *   XPIN1 = VCC
+ *   XPIN2 = uart0 [TX Pin]
+ *   XPIN3 = uart0 [RX Pin]
+ *   XPIN4 = <<UNUSED>>
+ *   XPIN5 = special0 [Reset Pin]
+ *   XPIN6 = special0 [RSSI PWM Pin]
+ *   XPIN7 = <<UNUSED>>
+ *   XPIN8 = special0 [BKGD Pin]
+ *   XPIN9 = power_management0 [Sleep Request Pin]
+ *   XPIN10 = GND
+ *   XPIN11 = <<UNUSED>>
+ *   XPIN12 = <<UNUSED>>
+ *   XPIN13 = power_management0 [On Sleep Pin]
+ *   XPIN14 = VCC REF
+ *   XPIN15 = special0 [Association Pin]
+ *   XPIN16 = <<UNUSED>>
+ *   XPIN17 = one_wire0 [1-Wire Bus pin]
+ *   XPIN18 = <<UNUSED>>
+ *   XPIN19 = <<UNUSED>>
+ *   XPIN20 = special0 [Commissioning Pin]
+ *
+ ************************************/
+
+#include <xbee_config.h>
+#include <types.h>
+#include <math.h>
+#include <xbee/transparent_serial.h>
+
+bool_t rx_ready = FALSE;
+bool_t at_ready = FALSE;
+int joined = 0;
+int32_t sleep_time = 0;
+
+/*
+ * Handles responses to the SN command
+ * 
+ * response: pointer to the packet struct containing the response
+ * 
+ * returns: XBEE_ATCMD_DONE to indicate to the dispatcher it is finished processing the packet
+ */
+int callback(const xbee_cmd_response_t FAR *response)
+{
+	if (response->flags & XBEE_CMD_RESP_FLAG_TIMEOUT || 
+	    response->flags & XBEE_AT_RESP_ERROR || 
+	    response->flags & XBEE_AT_RESP_BAD_PARAMETER || 
+	    response->flags & XBEE_AT_RESP_BAD_COMMAND)
+		at_ready = FALSE;
+	else {
+		sleep_time = (int)(response->value);

```

Comment:
why are we doing sleep time when setting it to be ready?

Reply:
at_ready is a variable that indicates a value has been received from an XBee AT parameter. Also, we're just storing the sleep time here, not actually sleeping the device.